lineapy/graph_reader/artifact_collection.py

```diff
@@ -42,6 +41,7 @@ def __init__(
         target_artifacts: List[LineaArtifactDef],
         input_parameters: List[str] = [],
         reuse_pre_computed_artifacts: List[LineaArtifactDef] = [],
+        dependencies: TaskGraphEdge = {},
```

Comment:
Migration of `dependencies` from `BasePipelineWriter` to `ArtifactCollection` makes implementation neater but is conceptually a bit awkward. Perhaps we may need to rethink/redefine what `ArtifactCollection` is.

Reply:
Ok I think conceptually we have a different idea on what `ArtifactCollection` is. I've personally felt its our internal representation of Pipeline, so any pipeline attributes which shouldnt change should also be reflected in its corresponding ArtifactCollection. 

If you feel strongly, lets discuss internally as a team.

Reply:
The initial intention, if I remember correctly, was to use it as a general collection of artifacts that can mutate into different components (one of which is the pipeline). Given that it has not found use outside pipeline-related applications, I am good with this scoping as long as we are clear about it (e.g., we need to update docstring for `ArtifactCollection` to clarify that it is an internal data structure to generate pipelines).